### PR TITLE
For #12831 - Disable SwipeRefreshLayout while swiping a bookmark.

### PR DESCRIPTION
**Pull request by person808**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by person808** [comment]
Makes sense. I'll fix up this PR so I can rebase #13046 on top of this one after

**Comment by person808** [comment]
@Mugurell ready for re-review!

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragmentStore.kt [hunk 1]
```diff
@@ -71,31 +74,37 @@ private fun bookmarkFragmentStateReducer(
...
+                BookmarkFragmentState.Mode.Selecting(items)
+            }
             state.copy(
-                mode = if (items.isEmpty()) {
-                    BookmarkFragmentState.Mode.Normal()
-                } else {
-                    BookmarkFragmentState.Mode.Selecting(items)
-                }
+                mode = mode,
+                isSwipeToRefreshEnabled = mode is BookmarkFragmentState.Mode.Normal
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragmentStore.kt:107]: For `SwipeRefreshAvailabilityChanged` and `Change` I see we're checking `mode !is BookmarkFragmentState.Mode.Selecting`.
To be consistent maybe we should do the same check here?

**Comment by person808** [comment]
> Just want to double check with you that we want to show pull to refresh if the mode is Syncing.

Yup! Otherwise the pull to refresh indicator won't show.

**Pull request merged**